src/Common/src/TypeSystem/Common/FieldDesc.FieldLayout.cs

```diff
@@ -38,17 +38,13 @@ public int Offset
         /// For static fields, represents whether or not the field is held in the GC or non GC statics region
         /// Does not apply to thread static fields.
         /// </summary>
-        public virtual bool HasGCStaticBase
+        public bool HasGCStaticBase
         {
             get
             {
                 Debug.Assert(IsStatic);
 
-                TypeDesc fieldType = FieldType;
-                if (fieldType.IsValueType)
-                    return ((DefType)fieldType).ContainsGCPointers;
-                else
-                    return fieldType.IsGCPointer;
+                return Context.ComputeHasGCStaticsBase(this);
```

Comment:
Consistency: The property is named `GCStaticBase`, but it calls into `GCStatic*s*Base`.